deployment/common/changeset/apply.go

```diff
@@ -0,0 +1,91 @@
+package changeset
+
+import (
+	"context"
+	"fmt"
+
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+type ChangesetApplication struct {
+	Changeset func(e deployment.Environment, config any) (deployment.ChangesetOutput, error)
+	Config    any
+}
```

Comment:
Can this use generics and be typesafe instead of `any`?

```golang
type ChangesetApplication[T any] struct {
	Changeset func(e deployment.Environment, config T) (deployment.ChangesetOutput, error)
	Config    T
}
```

Reply:
Also please add a comment for what this type's purpose is